lib/util/obstack.c

```diff
@@ -0,0 +1,490 @@
+/* obstack.c - subroutines used implicitly by object stack macros
+   Copyright (C) 1988, 89, 90, 91, 92, 93, 94 Free Software Foundation, Inc.
+
+This program is free software; you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation; either version 2, or (at your option) any
+later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
+#if HAVE_CONFIG_H
+# include <config.h>
+#if !defined HAVE_OBSTACK_H
+#include "hidrd/util/obstack.h"
+#endif
+#endif
+
+/* This is just to get __GNU_LIBRARY__ defined.  */
+#include <stdio.h>
+#include <stdlib.h>  //added A.Campbell contact@claydonkey.com fix implicit call to abort()
```

Comment:
required for MinGW Build but it might be needed by android. Haven't tested whether android requires it but there is  no harm in leaving it in